components/match2/wikis/tft/match_group_input_custom.lua

```diff
@@ -0,0 +1,185 @@
+	end
+
+	MatchFunctions.getTournamentVars(match)
+
+	match.stream = Streams.processStreams(match)
+	match.links = MatchFunctions.getLinks(match)
+
+	match.games = games
+	match.opponents = opponents
+
+	match.extradata = MatchFunctions.getExtraData(match)
+
+	return match
+end
+
+---@param match table
+---@return table
+function MatchFunctions.getTournamentVars(match)
+	match.mode = Variables.varDefault('tournament_mode', DEFAULT_MODE)
+	return MatchGroupInputUtil.getCommonTournamentVars(match)
+end
+
+---@param match table
+---@return table
+function MatchFunctions.getLinks(match)
+	return {
+		preview = match.preview
+	}
+end
+
+---@param match table
+---@return table
+function MatchFunctions.getExtraData(match)
+	return {
+		mvp = MatchGroupInputUtil.readMvp(match),
+		mapveto = MatchGroupInputUtil.getMapVeto(match, ALLOWED_VETOES),
+		casters = MatchGroupInputUtil.readCasters(match),
+	}
+end
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
+end
+
+---@param bestofInput string|integer?
+---@return integer?
+function MatchFunctions.getBestOf(bestofInput)
+	local bestof = tonumber(bestofInput) or tonumber(Variables.varDefault('match_bestof', DEFAULT_BESTOF))
+
+	if bestof then
+		Variables.varDefine('match_bestof', bestof)
+	end
+
+	return bestof
+end
+
+---@param match table
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	local maps = {}
+	for mapKey, map in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local finishedInput = map.finished --[[@as string?]]
+		local winnerInput = map.winner --[[@as string?]]
+
+		map.finished = MatchGroupInputUtil.mapIsFinished(map)
+
+		local opponentInfo = Array.map(Array.range(1, #opponents), function(opponentIndex)
+			local score, status = MatchGroupInputUtil.computeOpponentScore({
+				walkover = map.walkover,
+				winner = map.winner,
+				opponentIndex = opponentIndex,
+				score = map['score' .. opponentIndex],
+			}, MapFunctions.calculateMapScore(map.winner, map.finished))
+			return {score = score, status = status}
+		end)
+
+		map.scores = Array.map(opponentInfo, Operator.property('score'))
+		if map.finished then
+			map.resulttype = MatchGroupInputUtil.getResultType(winnerInput, finishedInput, opponentInfo)
+			map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInputUtil.getWinner(map.resulttype, winnerInput, opponentInfo)
+		end
+
+		map.extradata = MapFunctions.getExtradata(map, opponents)
+
+		table.insert(maps, map)
+		match[mapKey] = nil
+	end
+
+	return maps
+end
+
+---@param mapInput table
+---@param opponents table[]
+---@return table
+function MapFunctions.getExtradata(mapInput, opponents)
+	return {
+		comment = mapInput.comment,
+		header = mapInput.header,
```

Comment:
Is header needed?

Reply:
no idea i just ported the old pr^^

due to it being new i doubt it though

Reply:
no usage found. ia insource fwiw